docs/architecture/domain.md

```diff
@@ -0,0 +1,120 @@
+## Definition Model
+
+![Definition](assets/definition-model.png)
+
+### Source Code
+
+Source Code is a specified GitHub repository and branch that contains a `screwdriver.yaml` and the code required to build, test, and publish your application.
+
+### Step
+
+A step is a named action that needs to be performed, usually a single shell command. If the command finishes with a non-zero exit code, the step is considered a failure.
+
+### Container
+
+A Container runs [steps] in an isolated environment. This is done in order to test the compatibility of code running in different environments with different versions, without affecting other [tasks] that may be running at the same time. This is implemented using Docker containers.
+
+### Job
+
+A job consists of executing multiple sequential [steps] inside a specified [container]. If any step in the series fails, then the entire job is considered failed and subsequent steps will be skipped (unless configured otherwise).
+
+Jobs work by checking out the [source code] to a specified commit, setting the desired environment variables, and executing the specified [steps].
+
+During the job, the executing [steps] share three pieces of context:
```

Comment:
Eh, we didn't implement that yet.  It has some crazy side-effects (as we've seen).